client/block.c

```diff
@@ -1599,15 +1587,15 @@ int xdag_print_block_info(xdag_hash_t hash, FILE *out)
 	fprintf(out, "  file pos: %llx\n", (unsigned long long)bi->storage_pos);
 	fprintf(out, "      hash: %016llx%016llx%016llx%016llx\n",
 		(unsigned long long)h[3], (unsigned long long)h[2], (unsigned long long)h[1], (unsigned long long)h[0]);
-	fprintf(out, "    remark: %s\n", bi->remark ? bi->remark : "");
+	fprintf(out, "    remark: %s\n", ((bi->flags & BI_REMARK) && bi->bremark != NULL) ? (const char *)bi->bremark->remark : "");
 	fprintf(out, "difficulty: %llx%016llx\n", xdag_diff_args(bi->difficulty));
 	xdag_hash2address(h, address);
 	fprintf(out, "   balance: %s  %10u.%09u\n", address, pramount(bi->amount));
-	fprintf(out, "-------------------------------------------------------------------------------------------\n");
+	fprintf(out, "-----------------------------------------------------------------------------------------------------------------------------\n");
 	fprintf(out, "                               block as transaction: details\n");
 	fprintf(out, " direction  address                                    amount\n");
-	fprintf(out, "-------------------------------------------------------------------------------------------\n");
-	if(bi->ref) {
+	fprintf(out, "-----------------------------------------------------------------------------------------------------------------------------\n");
+	if((bi->flags & BI_REF) && bi->ref != NULL) {
```

Comment:
Bugfix bi->ref is not null because it refer to the orphan_block, need to check it is BI_REF too